azure/services/managedclusters/managedclusters.go

```diff
@@ -170,40 +121,58 @@ func (s *Service) IsManaged(ctx context.Context) (bool, error) {
   The token is used to create the admin kubeconfig.
   The user needs to ensure to provide service principle with admin aad privileges.
 */
-func (s *Service) ReconcileKubeconfig(ctx context.Context, managedClusterSpec azure.ResourceSpecGetter) (userKubeConfigData []byte, adminKubeConfigData []byte, err error) {
-	if s.Scope.IsAADEnabled() {
-		if userKubeConfigData, err = s.GetUserKubeconfigData(ctx, managedClusterSpec); err != nil {
+func reconcileKubeconfig(ctx context.Context, scope ManagedClusterScope, namespace string) (adminKubeConfigData []byte, userKubeConfigData []byte, err error) {
+	if scope.IsAADEnabled() {
+		if userKubeConfigData, err = getUserKubeconfigData(ctx, scope, namespace); err != nil {
 			return nil, nil, errors.Wrap(err, "error while trying to get user kubeconfig")
 		}
 	}
 
-	if s.Scope.AreLocalAccountsDisabled() {
-		userKubeconfigWithToken, err := s.GetUserKubeConfigWithToken(userKubeConfigData, ctx, managedClusterSpec)
+	if scope.AreLocalAccountsDisabled() {
+		userKubeconfigWithToken, err := getUserKubeConfigWithToken(userKubeConfigData, ctx, scope)
 		if err != nil {
 			return nil, nil, errors.Wrap(err, "error while trying to get user kubeconfig with token")
 		}
 		return userKubeconfigWithToken, userKubeConfigData, nil
 	}
 
-	adminKubeConfigData, err = s.GetCredentials(ctx, managedClusterSpec.ResourceGroupName(), managedClusterSpec.ResourceName())
+	asoSecret := &corev1.Secret{}
+	err = scope.GetClient().Get(
+		ctx,
+		client.ObjectKey{
+			Namespace: namespace,
+			Name:      adminKubeconfigSecretName(scope.ClusterName()),
+		},
+		asoSecret,
+	)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "failed to get credentials for managed cluster")
+		return nil, nil, errors.Wrap(err, "failed to get ASO admin kubeconfig")
 	}
+	adminKubeConfigData = asoSecret.Data[secret.KubeconfigDataName]
 	return adminKubeConfigData, userKubeConfigData, nil
 }
 
-// GetUserKubeconfigData gets user kubeconfig when aad is enabled for the aad clusters.
-func (s *Service) GetUserKubeconfigData(ctx context.Context, managedClusterSpec azure.ResourceSpecGetter) ([]byte, error) {
-	kubeConfigData, err := s.GetUserCredentials(ctx, managedClusterSpec.ResourceGroupName(), managedClusterSpec.ResourceName())
+// getUserKubeconfigData gets user kubeconfig when aad is enabled for the aad clusters.
+func getUserKubeconfigData(ctx context.Context, scope ManagedClusterScope, namespace string) ([]byte, error) {
```

Comment:
Bringing to attention: `getUserKubeconfigData()` is no more exported. So is `getUserKubeConfigWithToken()`. 
I think this is localized to this package and hence not exported?
Curious why were they exported earlier?

Reply:
My guess was that these were only exported before because the other methods that were already here also were. I made them unexported here to help clarify that these are not and should not be used outside this package.